File: arcade/0004-adjacentElementsProduct/adjacentElementsProduct.cpp
int adjacentElementsProduct(std::vector<int> inputArray) {
    size_t size = inputArray.size() - 1;
    int max = inputArray[0] * inputArray[1]; 
    
    for (size_t i = 1; i < size; ++i)
    {
        int product = inputArray[i] * inputArray[i + 1];

        if(product > max)
            max = product;
    }
    
    return max;
}

File: arcade/0004-adjacentElementsProduct/adjacentElementsProduct.swift
func adjacentElementsProduct(inputArray: [Int]) -> Int {
    var product = inputArray[0] * inputArray[1]
    
    for i in 1..<inputArray.count - 1 {
        let newProduct = inputArray[i] * inputArray[i + 1]
        
        if newProduct > product {
            product = newProduct
        }
    }
     
     return product
}
